xchange-bitfinex/src/main/java/org/knowm/xchange/bitfinex/service/BitfinexTradeService.java

```diff
@@ -186,44 +172,38 @@ public boolean cancelOrder(CancelOrderParams orderParams) throws IOException {
   @Override
   public UserTrades getTradeHistory(TradeHistoryParams params) throws IOException {
     try {
-      final String symbol;
+      String symbol = null;
       if (params instanceof TradeHistoryParamCurrencyPair
           && ((TradeHistoryParamCurrencyPair) params).getCurrencyPair() != null) {
         symbol =
             BitfinexAdapters.adaptCurrencyPair(
                 ((TradeHistoryParamCurrencyPair) params).getCurrencyPair());
-      } else {
-        // Exchange will return the errors below if CurrencyPair is not provided.
-        // field not on request: "Key symbol was not present."
-        // field supplied but blank: "Key symbol may not be the empty string"
-        throw new ExchangeException("CurrencyPair must be supplied");
       }
 
       long startTime = 0;
       Long endTime = null;
-      Integer limit = 50;
+      Long limit = 50L;
 
       if (params instanceof TradeHistoryParamsTimeSpan) {
         TradeHistoryParamsTimeSpan paramsTimeSpan = (TradeHistoryParamsTimeSpan) params;
-        startTime = DateUtils.toUnixTimeNullSafe(paramsTimeSpan.getStartTime());
-        endTime = DateUtils.toUnixTimeNullSafe(paramsTimeSpan.getEndTime());
+        startTime = DateUtils.toMillisNullSafe(paramsTimeSpan.getStartTime());
+        endTime = DateUtils.toMillisNullSafe(paramsTimeSpan.getEndTime());
       }
 
       if (params instanceof TradeHistoryParamPaging) {
```

Comment:
I think we should remove this entire `if` and the `pageNumber` from `BitfinexTradeHistoryParams` as this implementation has nothing in common with actual paging

Reply:
I agree: Done!